crates/router/src/core/webhooks.rs

```diff
@@ -389,6 +394,118 @@ pub async fn get_or_update_dispute_object(
+                    mandate_id.as_str(),
+                )
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::MandateNotFound)?,
+            webhooks::ObjectReferenceId::MandateId(
+                webhooks::MandateIdType::ConnectorMandateId(connector_mandate_id),
+            ) => db
+                .find_mandate_by_merchant_id_connector_mandate_id(
+                    &merchant_account.merchant_id,
+                    connector_mandate_id.as_str(),
+                )
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::MandateNotFound)?,
+            _ => Err(errors::ApiErrorResponse::WebhookProcessingFailure)
+                .into_report()
+                .attach_printable("received a non-mandate id for retrieving mandate")?,
+        };
+        let payment_intent = db
+            .find_payment_intent_by_payment_id_merchant_id(
+                &mandate
+                    .original_payment_id
+                    .ok_or(errors::ApiErrorResponse::InternalServerError)
+                    .into_report()
+                    .attach_printable("original_payment_id not present in mandate record")?,
+                &merchant_account.merchant_id,
+                merchant_account.storage_scheme,
+            )
+            .await
+            .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+        let payment_attempt = db
+            .find_payment_attempt_by_attempt_id_merchant_id(
+                &payment_intent.active_attempt_id.clone(),
+                &merchant_account.merchant_id,
+                merchant_account.storage_scheme,
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::PaymentNotFound)?;
+        let mandate_status = event_type
+            .foreign_try_into()
+            .into_report()
+            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
+            .attach_printable("event type to mandate status mapping failed")?;
+        let updated_mandate = db
+            .update_mandate_by_merchant_id_mandate_id(
+                &merchant_account.merchant_id,
+                &mandate.mandate_id,
+                storage::MandateUpdate::StatusUpdate { mandate_status },
+            )
+            .await
+            .to_not_found_response(errors::ApiErrorResponse::MandateNotFound)?;
+        db.update_payment_intent(
+            payment_intent.clone(),
+            storage::PaymentIntentUpdate::PGStatusUpdate {
+                status: api_models::enums::IntentStatus::Succeeded,
+            },
+            merchant_account.storage_scheme,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+        db.update_payment_attempt_with_attempt_id(
+            payment_attempt,
+            storage::PaymentAttemptUpdate::StatusUpdate {
+                status: api_models::enums::AttemptStatus::Charged,
```

Comment:
Why we are hard coding this as succeeded, this should be derived from mandate status.